controllers/machinedeletion_controller.go

```diff
@@ -63,19 +73,54 @@ func (r *MachineDeletionReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	if remediation = r.getRemediation(ctx, req); remediation == nil {
 		return ctrl.Result{}, nil
 	}
-	//not a machine based remediation
-	machineOwnerRef := getMachineOwnerRef(remediation)
-	if machineOwnerRef == nil {
-		return ctrl.Result{}, nil
+
+	var machine *unstructured.Unstructured
+	//Health check was done by NHC
+	if node, err := r.getNodeFromMdr(remediation); err == nil {
+		if machine, err = r.buildMachineFromNode(node); err != nil {
+			return ctrl.Result{}, err
+		}
+		if isMachineBelongToMasterNode(machine) {
+			return ctrl.Result{}, nil
+		}
+
+	} else { //Failed both in fetching the machine and in fetching the node
+		return ctrl.Result{}, err
 	}
-	//delete the machine
-	if err := r.deleteMachine(ctx, buildMachine(machineOwnerRef, remediation)); err != nil {
+
+	if err := r.deleteMachineOfNode(ctx, machine, remediation.Name); err != nil {
 		return ctrl.Result{}, err
```

Comment:
missing log message

Reply:
I'll skip this log since I'll log the failure cases inside the `deleteMachineOfNode`